src/sentry/incidents/endpoints/project_alert_rule_index.py

```diff
@@ -1,17 +1,77 @@
 from __future__ import absolute_import
 
+import six
+import math
+from datetime import datetime
 from copy import deepcopy
 
 from rest_framework import status
 from rest_framework.response import Response
 
+from django.utils import timezone
+
 from sentry import features
 from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.exceptions import ResourceDoesNotExist
 from sentry.api.paginator import OffsetPaginator
-from sentry.api.serializers import serialize
+from sentry.api.serializers import serialize, AlertRuleSerializer, CombinedRuleSerializer
 from sentry.incidents.models import AlertRule
-from sentry.incidents.endpoints.serializers import AlertRuleSerializer
+from sentry.models import Rule, RuleStatus
+from sentry.utils.cursors import build_cursor, Cursor
+
+
+class ProjectCombinedRuleIndexEndpoint(ProjectEndpoint):
+    def get(self, request, project):
+        """
+        Fetches alert rules and legacy rules for an organization
+        """
+        if not features.has("organizations:incidents", project.organization, actor=request.user):
+            raise ResourceDoesNotExist
+
+        cursor_string = request.GET.get("cursor", "0:0:0")
+        try:
+            limit = min(100, int(request.GET.get("limit", 25)))
+        except ValueError as e:
+            return Response(
+                {"detail": "Invalid input for `limit`. Error: %s" % six.text_type(e)}, status=400
+            )
+
+        cursor = Cursor.from_string(cursor_string)
+        cursor_date = datetime.fromtimestamp(float(cursor.value)).replace(tzinfo=timezone.utc)
+
+        alert_rule_queryset = (
+            AlertRule.objects.fetch_for_project(project)
+            .filter(date_added__gte=cursor_date)
+            .order_by("-date_added")[: limit + 1]
+        )
+
+        legacy_rule_queryset = (
+            Rule.objects.filter(
+                project=project, status__in=[RuleStatus.ACTIVE, RuleStatus.INACTIVE]
+            )
+            .select_related("project")
+            .filter(date_added__gte=cursor_date)
+            .order_by("-date_added")[: (limit + 1)]
+        )
+        combined_rules = list(alert_rule_queryset) + list(legacy_rule_queryset)
+        combined_rules.sort(
+            key=lambda instance: (instance.date_added, type(instance)), reverse=True
+        )
+        combined_rules = combined_rules[cursor.offset : cursor.offset + limit + 1]
+
+        def get_item_key(item, for_prev=False):
+            value = getattr(item, "date_added")
+            value = float(value.strftime("%s.%f"))
+            return math.floor(value)
```

Comment:
This could probably be one line? At least, we probably don't need `getattr`, and could just reference `item.date_added` directly?